ddl/db_partition_test.go

```diff
@@ -1891,11 +1891,13 @@ LOOP:
 			}
 			step := 10
 			rand.Seed(time.Now().Unix())
+			maxRetries := 3
```

Comment:
does the following the SQL can always be successfully in 3 times? 
maybe the failpoint is better?

Reply:
It should succeed in the second time. DDL won't change so fast.
It's not good to let the deletion succeed when there's a DDL change. So failpoint is not good.